hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/SparkHoodieTableFileIndex.scala

```diff
@@ -299,7 +299,9 @@ class SparkHoodieTableFileIndex(spark: SparkSession,
       // prefix to try to reduce the scope of the required file-listing
       val relativePartitionPathPrefix = composeRelativePartitionPath(staticPartitionColumnNameValuePairs)
 
-      if (staticPartitionColumnNameValuePairs.length == partitionColumnNames.length) {
+      if (!metaClient.getFs.exists(new Path(getBasePath, relativePartitionPathPrefix))) {
```

Comment:
`fs.exists` call is costly. This will impact latency. How often do we run into this scenario? FS cache is invalidated on each refresh anyway, so I am wondering if we really need to do fs.exists check everytime.
Can we not simply catch the exception and continue?

Reply:
Got it 👍

Reply:
we should avoid fs.exists call. all direct fs calls should get routed to BaseMetadataTable interface. if metadata is enabled, we fetch the value from metadata table.

Reply:
Does MDT has the interface to check whether a partition exists ? I don't think so.

Reply:
It does have getAllPartitionPaths but it is based in state of MDT and not fs.exists. If i understand correctly, this PR is to fix the issue when metadata is disabled and partitionPathPrefixAnalysis is enabled.